zkevm-circuits/src/sig_circuit/ecdsa.rs

```diff
@@ -100,36 +100,9 @@ where
     // - (u1 + u3) * G
     // - u2 * pubkey + u3 * G
     // are not equal
-    //
-    //     TODO: Technically they could be equal for a valid signature, but this happens with
-    // vanishing probability           for an ECDSA signature constructed in a standard way
-    // coordinates of u1_mul and u2_mul are in proper bigint form, and lie in but are not
-    // constrained to [0, n) we therefore need hard inequality here
     let u1_u2_x_eq = ecc_chip.is_equal(ctx, &u1u3_mul, &u2_pk_u3_g);
     let u1_u2_not_eq = base_chip.range.gate().not(ctx, Existing(u1_u2_x_eq));
 
-    // check u1*G and u2*pubkey are not negate of each other
```

Comment:
in halo2-lib's ecdsa circuit, they checked `(u1G).x != (u2pk).x`. 
this is an optimization for which they can avoid checking `(u1G).y != (u2pk).y`.
but this is also buggy as when `u1G == -u2 pk` it is also a valid signature.

In our refactor, we have already checked `u1G != u2 pk`. we do not need to check  `u1G ` and `u2 pk` are not negation of each other.